### ISSUE TITLE
Compact the gene and organism list tables

### ISSUE THREAD

**Issue by ValWood** [pombase/canto#1676]
Could you make the "genes" tables slightly more compact?

I only added  a single pathogen and a single host gene, and my screen is already full:

![compact this](https://user-images.githubusercontent.com/7359272/48481738-83979f80-e806-11e8-9b73-20bccf27ca89.jpg)

**Comment by jseager7** [comment]
Just to clarify, do you want the table to use less screen **width** or screen **height**, or both?

There's a few things I could try to minimise the organisms column. The width could be reduced by enforcing wrapping on some of the columns, but this does risk using more vertical space instead.

### Workflow

> I'm also not keen on the placement of the "add experimental strains for this organisms" in the table, or it's position in the workflow.

I think we did consider having the strain entry as a separate step in the workflow, such that the process would be something like:

Add genes and organisms &rarr; Confirm genes and organisms &rarr; Add strains

But it could also be:

Add genes and organisms &rarr; Add strains &rarr; Confirm genes and organisms

I can't remember the rationale for including it on this page, perhaps it was just in the interest of speed. 

### Strains and genes

> Certainly there needs to be some isolation from the gene...

The strains are only bound to the organism, but this might not have been clear because you only added a single gene for both species. If you added multiple genes, you would see that the strain picker only appears once for each organism:

![strain_organism_binding](https://user-images.githubusercontent.com/37659591/48483498-7e891f00-e80b-11e8-9133-4312f43a575e.PNG)

Admittedly it might not be too clear from the table layout what the strains are actually bound to. This could be solved by the workflow solutions above.

**Comment by ValWood** [comment]
> Just to clarify, do you want the table to use less screen width or screen height, or both?

height mainly

**Comment by ValWood** [comment]
strain selection display part moved to
https://github.com/pombase/canto/issues/1687

**Comment by kimrutherford** [comment]
> Reference Genome ID

To save some vertical space, I think that could probably be "ID" or "Identifier".

**Comment by ValWood** [comment]
> To save some vertical space, I think that could probably be "ID" or "Identifier".

That would be OK too...

**Comment by kimrutherford** [comment]
> Wouldn't
> Arabidopsis thaliana (Thale cress)
> be sufficient. 

Yep.  We could abbreviate the genus here too, especially it we include the common name:

A. thaliana (Thale cress)

**Comment by ValWood** [comment]
yes that would be sufficient, and it would compact these tables quite a lot

**Comment by ValWood** [comment]
taxon ID:
 don't know anyone who would depend on the taxon ID, or to who it would be critical. Most people don't know them...

**Comment by jseager7** [comment]
I've been working on this, and so far I've reduced the width of the headers and compacted the organism column:

![1676-compact-table](https://user-images.githubusercontent.com/37659591/51177990-3a70d400-18b9-11e9-8f5b-f41704bed86e.PNG)

Reducing the spacing between the tables is a bit more tricky &ndash; the styling rules that add the spacing are bundled up with all the rules that style the rest of the table, so it's awkward to change one without changing the other &ndash; I could override the rules, but another potential solution involves changing the table to use the same styling as other tables in Canto. Here's how that looks:

![1676-curs-style-organism-table](https://user-images.githubusercontent.com/37659591/51178175-b539ef00-18b9-11e9-88e7-7a4c17f5d70b.PNG)

However I think the row striping isn't ideal in this case. I could try and write new styling rules so that each organism row group has alternating colour. See the image below:

![1676-row-striping](https://user-images.githubusercontent.com/37659591/51178255-efa38c00-18b9-11e9-8d11-602475601586.PNG)

I think that makes it easier to distinguish organisms and their genes, but it could take a fair amount of work to implement this different system.

**Comment by jseager7** [comment]
> Reducing the spacing between the tables is a bit more tricky

Actually, it turned out this wasn't the case. The spacing was due to extra line breaks in the code. Here's how the spacing between the tables looks with the line breaks removed:

![1676-table-spacing](https://user-images.githubusercontent.com/37659591/51179104-8bce9280-18bc-11e9-979d-e0599ee8a848.PNG)

**Comment by jseager7** [comment]
I've now implemented the row striping so that the colours alternate by organism rather than by gene (that is, by row group rather than row). Here's a picture of the final layout:

![1676-row-striping-done](https://user-images.githubusercontent.com/37659591/51183999-e53dbe00-18ca-11e9-9f92-9b9fdd88f0ef.PNG)

I think the colour alternation might be in the opposite order of the row striping elsewhere (here the even rows are coloured instead of odd rows), but I thought this way looked clearer.

Note that I've also made some slight changes to the design of the text that shows when an organism has no genes. Here's how it looked before:

![1676-no-gene-before](https://user-images.githubusercontent.com/37659591/51184133-4feef980-18cb-11e9-897e-91d79312f0a0.PNG)

And here's how it looks now:

![1676-no-gene-after](https://user-images.githubusercontent.com/37659591/51184142-554c4400-18cb-11e9-8266-f379c8a16e8d.PNG)

My reason for the above change is that I thought it would make row groups with zero genes very visually distinct from rows with one gene. Previously I wasn't sure if the rows with zero genes stood out enough.

I'll open a pull request to make @kimrutherford aware of the technical side of these changes.
